diff --git a/render/egl.c b/render/egl.c
index 19868ca8..544b106b 100644
--- a/render/egl.c
+++ b/render/egl.c
@@ -440,61 +440,6 @@ static bool egl_init(struct wlr_egl *egl, EGLenum platform,
 
 static bool device_has_name(const drmDevice *device, const char *name);
 
-static EGLDeviceEXT get_egl_device_from_drm_fd(struct wlr_egl *egl,
-		int drm_fd) {
-	if (egl->procs.eglQueryDevicesEXT == NULL) {
-		wlr_log(WLR_DEBUG, "EGL_EXT_device_enumeration not supported");
-		return EGL_NO_DEVICE_EXT;
-	} else if (!egl->exts.EXT_device_query) {
-		wlr_log(WLR_DEBUG, "EGL_EXT_device_query not supported");
-		return EGL_NO_DEVICE_EXT;
-	}
-
-	EGLint nb_devices = 0;
-	if (!egl->procs.eglQueryDevicesEXT(0, NULL, &nb_devices)) {
-		wlr_log(WLR_ERROR, "Failed to query EGL devices");
-		return EGL_NO_DEVICE_EXT;
-	}
-
-	EGLDeviceEXT *devices = calloc(nb_devices, sizeof(*devices));
-	if (devices == NULL) {
-		wlr_log_errno(WLR_ERROR, "Failed to allocate EGL device list");
-		return EGL_NO_DEVICE_EXT;
-	}
-
-	if (!egl->procs.eglQueryDevicesEXT(nb_devices, devices, &nb_devices)) {
-		wlr_log(WLR_ERROR, "Failed to query EGL devices");
-		return EGL_NO_DEVICE_EXT;
-	}
-
-	drmDevice *device = NULL;
-	int ret = drmGetDevice(drm_fd, &device);
-	if (ret < 0) {
-		wlr_log(WLR_ERROR, "Failed to get DRM device: %s", strerror(-ret));
-		return EGL_NO_DEVICE_EXT;
-	}
-
-	EGLDeviceEXT egl_device = NULL;
-	for (int i = 0; i < nb_devices; i++) {
-		const char *egl_device_name = egl->procs.eglQueryDeviceStringEXT(
-				devices[i], EGL_DRM_DEVICE_FILE_EXT);
-		if (egl_device_name == NULL) {
-			continue;
-		}
-
-		if (device_has_name(device, egl_device_name)) {
-			wlr_log(WLR_DEBUG, "Using EGL device %s", egl_device_name);
-			egl_device = devices[i];
-			break;
-		}
-	}
-
-	drmFreeDevice(&device);
-	free(devices);
-
-	return egl_device;
-}
-
 static int open_render_node(int drm_fd) {
 	char *render_name = drmGetRenderDeviceNameFromFd(drm_fd);
 	if (render_name == NULL) {
@@ -524,24 +469,6 @@ struct wlr_egl *wlr_egl_create_with_drm_fd(int drm_fd) {
 		return NULL;
 	}
 
-	if (egl->exts.EXT_platform_device) {
-		/*
-		 * Search for the EGL device matching the DRM fd using the
-		 * EXT_device_enumeration extension.
-		 */
-		EGLDeviceEXT egl_device = get_egl_device_from_drm_fd(egl, drm_fd);
-		if (egl_device != EGL_NO_DEVICE_EXT) {
-			if (egl_init(egl, EGL_PLATFORM_DEVICE_EXT, egl_device)) {
-				wlr_log(WLR_DEBUG, "Using EGL_PLATFORM_DEVICE_EXT");
-				return egl;
-			}
-			goto error;
-		}
-		/* Falls back on GBM in case the device was not found */
-	} else {
-		wlr_log(WLR_DEBUG, "EXT_platform_device not supported");
-	}
-
 	if (egl->exts.KHR_platform_gbm) {
 		int gbm_fd = open_render_node(drm_fd);
 		if (gbm_fd < 0) {
diff --git a/types/output/output.c b/types/output/output.c
index 818f4549..cd713d82 100644
--- a/types/output/output.c
+++ b/types/output/output.c
@@ -886,8 +886,12 @@ const struct wlr_drm_format_set *wlr_output_get_primary_formats(
 }
 
 bool wlr_output_is_direct_scanout_allowed(struct wlr_output *output) {
+	static bool warned = false;
 	if (output->attach_render_locks > 0) {
-		wlr_log(WLR_DEBUG, "Direct scan-out disabled by lock");
+		if (!warned) {
+			wlr_log(WLR_DEBUG, "Direct scan-out disabled by lock");
+			warned = true;
+		}
 		return false;
 	}
 
@@ -896,8 +900,11 @@ bool wlr_output_is_direct_scanout_allowed(struct wlr_output *output) {
 	wl_list_for_each(cursor, &output->cursors, link) {
 		if (cursor->enabled && cursor->visible &&
 				cursor != output->hardware_cursor) {
-			wlr_log(WLR_DEBUG,
-				"Direct scan-out disabled by software cursor");
+			if (!warned) {
+				wlr_log(WLR_DEBUG,
+					"Direct scan-out disabled by software cursor");
+				warned = true;
+			}
 			return false;
 		}
 	}

