diff --git a/common/Makefile.am b/hybris/common/Makefile.am
index 7cd5880..59dd64a 100644
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -114,3 +114,6 @@ hooks_libhybris_hooks_la_SOURCES = \
 hooks_libhybris_hooks_la_CXXFLAGS = \
 	$(libhybris_common_la_CXXFLAGS) \
 	-fvisibility=hidden
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = hybris-common.pc
diff --git a/hybris/common/hybris-common.pc.in b/hybris/common/hybris-common.pc.in
new file mode 100644
index 0000000..daf4586
--- /dev/null
+++ b/common/hybris-common.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+includedir=@includedir@
+
+Name: hybris-common
+Description: libhybris common library
+Version: @VERSION@
+Libs: -L${libdir} -lhybris-common
+Cflags: -I${includedir}
diff --git a/hybris/configure.ac b/hybris/configure.ac
index 6d02935..c691b64 100644
--- a/configure.ac
+++ b/configure.ac
@@ -304,6 +304,7 @@ AC_CONFIG_FILES([
 	camera/Makefile
 	camera/libcamera.pc
 	common/Makefile
+	common/hybris-common.pc
 	common/jb/Makefile
 	common/stub/Makefile
 	common/mm/Makefile
