diff --git a/hybris/common/q/linker.cpp b/hybris/common/q/linker.cpp
index dcc8428..78e3d76 100644
--- a/common/q/linker.cpp
+++ b/common/q/linker.cpp
@@ -4261,6 +4261,7 @@ std::vector<android_namespace_t*> init_default_namespaces(const char* executable
 
   std::string error_msg;
 
+#if 0
   std::string ld_config_file_path = get_ld_config_file_path(executable_path);
 
   INFO("[ Reading linker config \"%s\" ]", ld_config_file_path.c_str());
@@ -4277,6 +4278,7 @@ std::vector<android_namespace_t*> init_default_namespaces(const char* executable
     }
     config = nullptr;
   }
+#endif
 
   if (config == nullptr) {
     return init_default_namespace_no_config(g_is_asan);

